plugins/fluentd_telemetry_plugin/src/streamer.py

```diff
@@ -702,44 +865,59 @@ def stream_data(self, telemetry_endpoint):  # pylint: disable=too-many-locals
         _port = telemetry_endpoint.get(self.config_parser.UFM_TELEMETRY_ENDPOINT_SECTION_PORT)
         _url = telemetry_endpoint.get(self.config_parser.UFM_TELEMETRY_ENDPOINT_SECTION_URL)
         msg_tag = telemetry_endpoint.get(self.config_parser.UFM_TELEMETRY_ENDPOINT_SECTION_MSG_TAG_NAME)
+        is_xdr_mode = telemetry_endpoint.get(self.config_parser.UFM_TELEMETRY_ENDPOINT_SECTION_XDR_MODE)
         telemetry_data = self._get_metrics(_host, _port, _url, msg_tag)
-        if telemetry_data:
-            try:
+        try:
+            data_to_stream = []
+            new_data_timestamp = None
+            num_of_counters = data_len = 0
+            if telemetry_data:
                 ufm_telemetry_is_prometheus_format = self._check_data_prometheus_format(telemetry_data)
                 logging.info('Start Processing The Received Response From %s', msg_tag)
                 start_time = time.time()
-                data_to_stream, new_data_timestamp, num_of_counters = self._parse_telemetry_prometheus_metrics_to_json(telemetry_data) \
+                data_to_stream, new_data_timestamp, num_of_counters = \
+                    self._parse_telemetry_prometheus_metrics_to_json(telemetry_data, msg_tag) \
```

Comment:
There is no Prometheus with XDR?

Reply:
Currently, there are some limitations with using XDR in the Prometheus format, as detailed in [#4096650](https://redmine.mellanox.com/issues/4096650). 
Additionally, in the context of TFS, the Prometheus format is not commonly used. Most use cases prefer the CSV format due to its smaller response size compared to Prometheus.